hcx-apis/src/main/java/org/swasth/hcx/controllers/BaseController.java

```diff
@@ -94,4 +97,22 @@ private void processAndSendEvent(String apiAction, String metadataTopic, Map<Str
             return new ResponseEntity<>(errorResponse(response, ErrorCodes.SERVER_ERROR, e), HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
+
+    public Map<String, Object> formatsRequestBody(Map<String, Object> requestBody) {
+        Map<String, Object> event = new HashMap<>();
+        try {
+            String str = (String) requestBody.get("payload");
+
+            String[] strArray = str.split("\\.");
```

Comment:
You have to check the size of the array and proceed.

Reply:
Done